dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java

```diff
@@ -0,0 +1,313 @@
+        
+        testSubFolder =  new FolderDataGen().name("testSubFolder").parent(testFolder).nextPersisted();
+
+        Template template=new Template();
+        template.setTitle("a template "+UUIDGenerator.generateUuid());
+        template.setBody("<html><body> I'm mostly empty </body></html>");
+        template=APILocator.getTemplateAPI().saveTemplate(template, testHost, APILocator.systemUser(), false);
+        
+        HTMLPageAsset page = new HTMLPageDataGen(testFolder, template).next();
+        page.setTitle("testingpage1");
+        testPage = APILocator.getHTMLPageAssetAPI().fromContentlet(HTMLPageDataGen.checkin(page, IndexPolicy.FORCE));
+
+        testlink = new LinkDataGen().hostId(testHost.getIdentifier()).title("testLink").parent(testFolder).target("https://google.com").linkType("EXTERNAL").nextPersisted();
+
+    }
+
+    
+
+
+
+    
+    @Test(expected = NotFoundInDbException.class)
+    public void testGetFolderContentWithInvalidIdentifier() throws DotDataException, DotSecurityException { // https://github.com/dotCMS/core/issues/11829
+
+        final String NOT_EXISTING_ID = "01234567-1234-1234-1234-123456789012";
+
+        browserAPI.getFolderContent( APILocator.systemUser(), NOT_EXISTING_ID, 0, -1, "", null, null, true, false, false, false, "", false, false, 1 );
+    }
+
+    @Test
+    public void testGetFolderContentWithValidIdentifier() throws Exception { // https://github.com/dotCMS/core/issues/11829
+
+        final String folderPath = "/BrowserAPITest-Folder";
+
+        //Creating folder to check.
+        User user = userAPI.getSystemUser();
+        Host demo = testHost;
+        Folder folder = folderAPI.createFolders( folderPath, demo, user, false );
+
+        try {
+            Map<String, Object> folderContent = browserAPI.getFolderContent( APILocator.systemUser(), folder.getInode(), 0, -1, "", null, null, true, false, false, false, "", false, false, 1 );
+            Assert.assertTrue( folderContent.containsKey( "total" ) );
+            Assert.assertTrue( folderContent.containsKey( "list" ) );
+        } catch ( Exception e ){
+            Assert.fail( "We should not be getting any exception here" );
+        } finally {
+            folderAPI.delete( folder, user, false );
+        }
+    }
+    
+    
+    
+    
+    @Test
+    public void testingDifferentBrowserAPIResults() throws Exception{
+        for(Tuple3<String,BrowserQuery, Set<String>> testCase : browserApiTestCases()) {
+            String testTitle = testCase._1;
+            Map<String,Object> results = browserAPI.getFolderContent(testCase._2);
+            
+            List<String> list = ((List<Map<String,Object>>) results.get("list")).stream().map(m->(String) m.get("name")).collect(Collectors.toList());
+            
+            assertTrue(testTitle, !list.isEmpty());
+            
+            Set<String> expectedNames = testCase._3;
+            
+            assertTrue(testTitle, list.size()==expectedNames.size());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use assertSame(x, y) instead of assertTrue(x==y), or assertNotSame(x,y) vs assertFalse(x==y)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)